### ISSUE TITLE
Several quirks in YAML rendering (colon in attribute name)

### ISSUE THREAD

**Issue by marians** [highlightjs/highlight.js#1681]
I'd like to report a few problems we are seeing in YAML rendering.

Here is my reference document:

```yaml
endpoints:
  http://localhost:9000:
    alias: local
    email: developer@example.com
    token: 7d5b1161-87e6-4dfd-8017-5ab9f4e1cb14
  https://api.g8s.cloud.europe.gigantic.io:
    alias: cloud
    email: developer@example.com
    token: 103dc680-1b53-4ad3-b5cb-ac9b2f86134d
  https://api.g8s.onprem.example.com:
    alias: onprem
    email: developer@example.com
    token: ""
selected_endpoint: https://api.g8s.cloud.europe.gigantic.io
last_version_check: 2018-01-25T13:50:25.353937+01:00
updated: 2018-01-26T09:17:24+01:00
```

That YAML is generated using [go-yaml v2](https://github.com/go-yaml/yaml/tree/v2).

This is how highlight.js version 9.12.0 renders it with solarized-dark style:

![image](https://user-images.githubusercontent.com/273727/35444839-10d384a0-02b0-11e8-8444-3663983b4238.png)

The HTML code for the above:

```html
<pre><code class="language-yaml hljs"><span class="hljs-attr">endpoints:</span>
<span class="hljs-attr">  http:</span><span class="hljs-string">//localhost:9000:</span>
<span class="hljs-attr">    alias:</span> <span class="hljs-string">local</span>
<span class="hljs-attr">    email:</span> <span class="hljs-string">developer@example.com</span>
<span class="hljs-attr">    token:</span> <span class="hljs-number">7</span><span class="hljs-string">d5b1161-87e6-4dfd-8017-5ab9f4e1cb14</span>
<span class="hljs-attr">  https:</span><span class="hljs-string">//api.g8s.cloud.europe.gigantic.io:</span>
<span class="hljs-attr">    alias:</span> <span class="hljs-string">cloud</span>
<span class="hljs-attr">    email:</span> <span class="hljs-string">developer@example.com</span>
<span class="hljs-attr">    token:</span> <span class="hljs-number">103</span><span class="hljs-string">dc680-1b53-4ad3-b5cb-ac9b2f86134d</span>
<span class="hljs-attr">  https:</span><span class="hljs-string">//api.g8s.onprem.example.com:</span>
<span class="hljs-attr">    alias:</span> <span class="hljs-string">onprem</span>
<span class="hljs-attr">    email:</span> <span class="hljs-string">developer@example.com</span>
<span class="hljs-attr">    token:</span> <span class="hljs-string">""</span>
<span class="hljs-attr">selected_endpoint:</span> <span class="hljs-attr">https://api.g8s.cloud.europe.gigantic.io</span>
<span class="hljs-attr">last_version_check:</span> <span class="hljs-number">2018</span><span class="hljs-bullet">-01</span><span class="hljs-bullet">-25</span><span class="hljs-attr">T13:50:25.353937+01:00</span>
<span class="hljs-attr">updated:</span> <span class="hljs-number">2018</span><span class="hljs-bullet">-01</span><span class="hljs-bullet">-26</span><span class="hljs-attr">T09:17:24+01:00</span>
</code></pre>
```

**Comment by marians** [comment]
Starting from the top, the first obvious mistake is that the colon `:` is not expected to be part of the attribute name. Perhaps the later mistakes are all based on that one.

**Comment by Gwend4l** [comment]
There is also a problem when there is a full stop in the middle of a key.

Test case:
```yaml
root:
  key: "value"
  other.key: "other value"
```

Result with highlight.js:
![capture d ecran de 2018-02-20 16-56-40](https://user-images.githubusercontent.com/1388961/36434419-421e7f16-165f-11e8-955d-fe622b76acea.png)
`other` should be highlighted as well

**Comment by Gwend4l** [comment]
Having a small dict (`key : value`) in a list item also break highlight.js.

Test case :
```yaml
root:
  - key1: "value1"
  - key2: "value2"
```
Result with highlight.js:
![capture d ecran de 2018-02-20 17-18-33](https://user-images.githubusercontent.com/1388961/36435555-43e2138c-1662-11e8-897e-33f3368e4986.png)
`root`, `key1` and `key2` should all be highlighted the same (in blue in my color theme)

Comparison with a dict without a list:
![capture d ecran de 2018-02-20 17-18-29](https://user-images.githubusercontent.com/1388961/36435589-56bf6036-1662-11e8-8ac1-b311db481fc8.png)

**Comment by cdbax** [comment]
This was fixed by #2170 but I didn't do the auto-linking to close it properly

**Comment by yyyc514** [comment]
Yes, if you can test against `master` this should be resolved now.
